### PR TITLE
Getting version numbers updated automatically

### PR DESCRIPTION
**Pull request by yhakbar**
Hopefully, this will do for now, to keep the version in the README.md updated to the latest version.

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -2,14 +2,33 @@
...
+    if is_gnu_sed; then
+        sed -i -r "$EXPRESSION" "$FILE"
+    else
+        sed -i '' -E "$EXPRESSION" "$FILE"
+    fi
+}
+
 cd "$( dirname "${BASH_SOURCE[0]}" )/.."
 
 CURRENT=$(grep -E -o '[0-9]' ./internal/gorson/version/version.go)
```
Comment (joshfinnie) [inline scripts/release.sh:21]: this is pretty interesting... this doesn't work once we get to version 10

```
1
0
```

we should update this to:

```
CURRENT=$(grep -E -o '[0-9].' ./internal/gorson/version/version.go)
```

Comment (yhakbar) [reply scripts/release.sh]: Good catch! I changed it to:

```bash
CURRENT=$(grep -E -o '[0-9]+' ./internal/gorson/version/version.go)
```

as that captures any natural number version without capturing quotes.

**Review by joshfinnie — APPROVED** [review]

**Comment by yhakbar** [comment]
@joshfinnie I think the most recent commit here resolves the weird binary naming we saw on [that release](https://github.com/pbs/gorson/releases/tag/11) that was created earlier? I'm not sure why it's called `1.1` instead of just `1`, though.

**Review by joshfinnie — APPROVED** [review]

**Review by joshfinnie — CHANGES_REQUESTED** [review]
The release binary was misnamed on a accidental run of this script. Let's figure out why...

**Review by joshfinnie — APPROVED** [review]

**Pull request merged**